### PR TITLE
Fix for issue #2218

### PR DESCRIPTION
**Pull request by Nakkar**
Fixed issue where Redstone didn't properly interact with transparent blocks.

### CODE REVIEW

**Comment by Schwertspize** [comment]
CircleCI failed, that means your code doesn't meet our conventions:

````Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h(51): 1 .. 36: Indenting with a space
Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h(52): 146 .. 149: Trailing whitespace````

File: src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ class cRedstoneWireHandler : public cRedstoneHandler
 	const cVector3iArray GetTerracingConnectionOffsets(const Vector3i & a_Position)
 	{
 		cVector3iArray RelativePositions;
-		bool IsYPTerracingBlocked = cBlockInfo::IsSolid(m_World.GetBlock(a_Position + OffsetYP()));
+		bool IsYPTerracingBlocked = cBlockInfo::IsSolid(m_World.GetBlock(a_Position + OffsetYP())) ? !cBlockInfo::IsTransparent(m_World.GetBlock(a_Position + OffsetYP())) : false;
```
Comment (madmaxoft) [inline src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h]: Wouldn't a simple `&&` between the two properties work as intended?
`IsSolid() && !IsTransparent()`

**Comment by SphinxC0re** [comment]
Please change your commits message to something more describing

File: src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h [hunk 2]
```diff
@@ -48,7 +48,8 @@ class cRedstoneWireHandler : public cRedstoneHandler
 			}
 
 			if (
-				!cBlockInfo::IsSolid(m_World.GetBlock(a_Position + Adjacent)) &&  // IsYMTerracingBlocked (i.e. check block above lower terracing position, a.k.a. just the plain adjacent)
+                                    // IsYMTerracingBlocked (i.e. check block above lower terracing position, a.k.a. just the plain adjacent)
+				(!cBlockInfo::IsSolid(m_World.GetBlock(a_Position + Adjacent)) ? true : cBlockInfo::IsTransparent(m_World.GetBlock(a_Position + Adjacent))) &&
```
Comment (madmaxoft) [inline src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h]: Again, `||` should be more elegant

**Comment by madmaxoft** [comment]
This PR has been replaced by #2218 , closing.

**Pull request closed without merging**